### DISCUSSION TITLE
Clarification about the scripts please

### DISCUSSION THREAD

**Discussion by TheCableGuy99** [007revad/Synology_HDD_db#81] [Q&A] [answered]
Hi,

So firstly, I have a DS1821+ with a DS517 extension bay with the following config:

-=DS1821+=-
WDC WD60EFRX (6TB) x5
WDC WD60EFAX (6TB) x2
WDC WD60EFZX (6TB) x1

-=DS1821+ M.2 Slots=-
Samsung Evo 970 Evo Plus SSD (1TB) x2

-=DS518=-
Seagate  ST8000DM0004 (8TB) x4

I wanted to use the M.2 slots as drives rather than cache so upgraded to DSM 7.2 Beta (which has now been upgraded automatically to 7.2 public release).

I installed the syno_hdd_db.sh script and configured it with a task to run automatically on boot.

Now my question is...
As I understand it, the syno_hdd_db.sh script updates the database of approved/supported drives so the NAS is able to run my non-official M.2 drives as a drive rather than cache. Is this correct? If not, please could you confirm what I have mis-understood? And either way, does it need to be run each time at boot?

My next question is:
What is the script Synology_enable_M2_volume for? The description comes across as it allows you to run M.2 drives with unsupported drives. But this is what I thought the syno_hdd_db.sh script does so am confused?

Lastly:
What is the script Synology_M2_volume for? Again, it just comes across as doing the exact same thing and is confusing to me.

Thanks for your help.

**Comment by 007revad** [comment, marked as answer]
Synology_M2_volume and Synology_enable_M2_volume were written before Synology_HDD_db was updated to enable creating M2 storage pools.

The preferred script to use is Synology_HDD_db and it is all you need with a DS1821+ using DSM 7.1.1 or 7.2

The 2 M2_volume scripts would only be used if Synology_HDD_db wasn't able to enable creating M2 storage pools in storage manager. 
And both M2_volume scripts need Synology_HDD_db as well.

Synology_enable_M2_volume changes a file in DSM so storage manager thinks all your drives are Synology drives. This allows you to create the storage pool and volume all from storage manager. It only works for DSM 7.2 (and DSM 7.1.1 on some NAS models) that already support M2 volumes. RAID repair and expansion are available in storage manager. Synology_enable_M2_volume is no longer needed as Synology_HDD_db now achieves the same result in a more reliable way.

Synology_M2_volume is only really needed for DSM 6 to 7.1.1 or NAS models that don't support M2 volumes even with DSM 7.2. It creates the system and data partitions, and the storage pool so you can create the volume in storage manager. For DSM 6 it also creates the volume. SHR and JBOD are not available. RAID repair and expansion are NOT available.

**Comment by TheCableGuy99** [comment]
Hi mate,

Thanks so much for taking the time to clarify all that for me. I have just 2 questions before finishing this topic if you don't mind?

1) Does it need to be run every time on boot? If not how often should it be re-run?
2) Is there a way to automated the updating of the script?

Thanks again, really appreciate it :)

**Comment by 007revad** [comment]
Synology_HDD_db needs to run after each DSM update. It doesn't hurt to run it every time on boot because it doesn't do anything unless it needs to.

Your second question is interesting because nobody has asked it before. I generally don't likes things updating automatically in case the new version has a bug. I like to wait a few days before updating.

I could add an option to auto-update the script x days after the new version was released. This would be useful for people who have the script scheduled.

**Comment by TheCableGuy99** [comment]
Thanks for clarifying that.

I just like automation, I have so much to keep track of anyway with my home network in regards to updates and maintenance that it's nice to automate things.

I appreciate things have bugs from time-to-time but personally I would like to see the option if you decide to add it.  I won't lose sleep over it if you device not to though.

Many thanks for answering my questions (and the script itself), really appreciate it :)

**Comment by 007revad** [comment]
I have new pre-release version that includes an auto update option. It also reloads the script automatically after it updates itself.
https://github.com/007revad/Synology_HDD_db/releases/tag/v2.3.48-RC

If you want to test it you'd need to edit `scriptver="v2.3.48"` on line 157 to a lower version.

The --autoupdate option requires a number for the number of days to wait after a new version is released before downloading it. You can set it to 0 to update right away.

For example:
`--autoupdate=0` would update right away.
`--autoupdate=1` would update 1 day after the version was released.
`--autoupdate=2` would update 2 days after the version was released.
